mesonbuild/compilers/objcpp.py

```diff
@@ -96,7 +96,9 @@ def get_options(self) -> 'coredata.MutableKeyedOptionDictType':
         opts.update({
             OptionKey('std', machine=self.for_machine, lang='cpp'): coredata.UserComboOption(
                 'C++ language standard to use',
-                ['none', 'c++98', 'c++11', 'c++14', 'c++17', 'gnu++98', 'gnu++11', 'gnu++14', 'gnu++17'],
+                ['none', 'c++98', 'c++11', 'c++14', 'c++17', 'c++20', 'c++2b',
+                 'gnu++98', 'gnu++11', 'gnu++14', 'gnu++17', 'gnu++20',
+                 'gnu++2b'],
```

Comment:
Do we want / need the “2b” versions in here?  It seems like the convention in C++ is to no longer add them to the cpp_std but rely on the fallback.

Reply:
Good question, should we prefer `c++23` + `gnu++23` instead?

(Note that the compiler itself currently only recognizes `c++2b`/`gnu++2b`, so we would always be doing the fallback. But that's probably fine.)

Reply:
FWICT whatever values for the version we have for C++ should also be used for Objective C++. If someone can come up with a counterexample, do speak up. Otherwise let's just go with feature parity.